### PR TITLE
Implement common codebuild generator/runner

### PR DESCRIPTION
**Pull request by ColdenCullen**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild/common.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: 0.2
+#this buildspec assumes the ubuntu 14.04 trusty image
```
Comment (justinboswell) [inline codebuild/common.yml]: This isn't true anymore, surely :)

File: codebuild/builder.py [hunk 2]
```diff
@@ -0,0 +1,693 @@
...
+        'account': github_account,
+        'spec': config['spec'].name(),
+    }
+
+    # This matches the CodeBuild API for expected format
+    CREATE_PARAM_TEMPLATE = {
+        'name': '{project}-{spec}',
+        'source': {
+            'type': 'GITHUB',
+            'location': 'https://github.com/{account}/{project}.git',
```
Comment (JonathanHenson) [inline codebuild/builder.py]: this is good, helps the GA switch be much smoother.

File: codebuild/common.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+version: 0.2
+phases:
+  install:
+    commands:
+# To be uncommented when the file is properly updated in master
+#      - python3 -c "from urllib.request import urlretrieve; urlretrieve('https://raw.githubusercontent.com/awslabs/aws-c-common/master/codebuild/builder.py', 'builder.py')"
```
Comment (JonathanHenson) [inline codebuild/common.yml]: Can we just rebuild the images to have elasticurl?

Comment (ColdenCullen) [reply codebuild/common.yml]: Paging mr @justinboswell

File: codebuild/builder.py [hunk 4]
```diff
@@ -0,0 +1,731 @@
...
+########################################################################################################################
+
+def run_build(build_spec, is_dryrun):
+
+    if not is_dryrun:
+        import tempfile, shutil, subprocess
+
+    #TODO These platforms don't succeed when doing a RelWithDebInfo build
+    if build_spec.host in ("al2012", "manylinux"):
+        global BUILD_CONFIG
```
Comment (justinboswell) [inline codebuild/builder.py]: Why not just cache it into a local and not mess with the global?

Comment (ColdenCullen) [reply codebuild/builder.py]: This way when we fix these issues, I can just nuke this chunk and the problem goes away

**Pull request merged**